### ISSUE TITLE
add desktop launchers for opening nautilus windows to grid computers

### ISSUE THREAD

**Issue by stefco** [stefco/geco_vm#65]
```
nautilus sftp://c1/home/$(ligo-auth-get-username)
```

where .ssh/config looks like

```
Host c1
    HostName ldas-pcdev1.ligo.caltech.edu
    ControlMaster auto
    ControlPath ~/.ssh/control:%h:%p:%r
```

http://unix.stackexchange.com/questions/33557/using-an-already-established-ssh-channel

so would want to:
- [x] make .ssh/config hosts for several datagrid computers
- [x] chmod 600 the config file (and make sure it belongs to vagrant)
- [x] make a script that opens such a connection with `gsissh` background master connections (like in the medm_lho and medm_llo scripts) before opening the nautilus windows
- [x] put these as right-click options in a central launcher for browsing your remote files, as in [these instructions](http://askubuntu.com/questions/13758/how-can-i-edit-create-new-launcher-items-in-unity-by-hand)

**Comment by stefco** [comment]
http://askubuntu.com/questions/13758/how-can-i-edit-create-new-launcher-items-in-unity-by-hand

**Comment by stefco** [comment]
UWM clusters:
pcdev1.cgca.uwm.edu
pcdev2.cgca.uwm.edu
pcdev3.cgca.uwm.edu

CalTech:
ldas-grid.ligo.caltech.edu
ldas-pcdev1.ligo.caltech.edu
ldas-pcdev2.ligo.caltech.edu
ldas-pcdev3.ligo.caltech.edu
ldas-pcdev4.ligo.caltech.edu
ldas-pcdev5.ligo.caltech.edu
ldas-pcdev6.ligo.caltech.edu
ldas-pcdev7.ligo.caltech.edu
ldas-pcdev8.ligo.caltech.edu
ldas-pcdev9.ligo.caltech.edu
ldas-pcdev10.ligo.caltech.edu

LHO:
ldas-grid.ligo-wa.caltech.edu
ldas-pcdev1.ligo-wa.caltech.edu
ldas-pcdev2.ligo-wa.caltech.edu

LLO:
ldas-grid.ligo-la.caltech.edu
ldas-pcdev1.ligo-la.caltech.edu
ldas-pcdev2.ligo-la.caltech.edu

**Comment by stefco** [comment]
fill the .ssh/config file and the desktop launcher with ligo hosts. call this provisioner `nautilus-gsissh-config`:

``` bash
mkdir -p "${HOME}"/.ssh

hosts=(
    pcdev1.cgca.uwm.edu
    pcdev2.cgca.uwm.edu
    pcdev3.cgca.uwm.edu
    ldas-grid.ligo.caltech.edu
    ldas-pcdev1.ligo.caltech.edu
    ldas-pcdev2.ligo.caltech.edu
    ldas-pcdev3.ligo.caltech.edu
    ldas-pcdev4.ligo.caltech.edu
    ldas-pcdev5.ligo.caltech.edu
    ldas-pcdev6.ligo.caltech.edu
    ldas-pcdev7.ligo.caltech.edu
    ldas-pcdev8.ligo.caltech.edu
    ldas-pcdev9.ligo.caltech.edu
    ldas-pcdev10.ligo.caltech.edu
    ldas-grid.ligo-wa.caltech.edu
    ldas-pcdev1.ligo-wa.caltech.edu
    ldas-pcdev2.ligo-wa.caltech.edu
    ldas-grid.ligo-la.caltech.edu
    ldas-pcdev1.ligo-la.caltech.edu
    ldas-pcdev2.ligo-la.caltech.edu)

# make the .ssh/config file for these hosts to force shared connections
# (this is what allows nautilus browsing)
for host in ${hosts[*]}; do
    echo "Host ${host}" >> "${HOME}"/.ssh/config
    echo "    HostName ${host}" >> "${HOME}"/.ssh/config
    echo "    ControlMaster auto" >> "${HOME}"/.ssh/config
    echo "    ControlPath ~/.ssh/control:%h:%p:%r" >> "${HOME}"/.ssh/config
done

chown -R vagrant "${HOME}"/.ssh
chmod 600 "${HOME}"/.ssh/config

# Add these as right-click options to the desktop launcher
app="${HOME}"/.medm_launcher/browse-data-grid.desktop

# The name of each action is just the host URL:
echo "Actions="$(sed 's/ /;/g' <<<"${hosts[@]}") >> "${app}"

# create an action for each of these host URLs
for host in ${hosts[*]}; do
    echo >> "${app}"
    echo "[Desktop Action ${host}]"  >> "${app}"
    echo "Name=Browse Home Folder on ${host}" >> "${app}"
    echo "Exec=nautilus-gsissh ${host}" >> "${app}"
done
```

the desktop launcher (`browse-data-grid.desktop`):

```
[Desktop Entry]
Type=Application
Terminal=false
Name=Browse Home Folder on Remote DataGrid Computer
Icon=/home/vagrant/.medm_launcher/browse-data-grid.png
Exec=zenity --info --text="Right click on the DataGrid icon and pick a server to connect to."
```

**Comment by stefco** [comment]
seems to be working based on testing.

**Comment by stefco** [comment]
implemented in d5e5c0c
